android/KMEA/app/src/main/java/com/tavultesoft/kmea/LanguageListActivity.java

```diff
@@ -640,7 +640,11 @@ public void onClick(DialogInterface dialog,int id) {
       });
 
     alertDialog = alertDialogBuilder.create();
-    alertDialog.show();
+    if (!((AppCompatActivity)context).isFinishing()) {
```

Comment:
Do you even need to create the Alert dialog if `isFinishing()` is true?

Reply:
The activity could have finished between the time the Alert dialog is created and shown.
From the reference links, we just need to check before the dialog is shown.

Reply:
I see. Does this mean there is still a race condition here?

Reply:
I don't think that's the case. We should be able to place this test at the very start of the function because this runs in the UI thread context, so there is no possibility of the activity finishing in the middle of the function.

The issue arises because this is called from a callback, so in fact the best place to do this is even earlier, before we attempt to show the error dialog, in `onKeyboardDownloadFinished`.

Reply:
ok - I've moved the activity check to `onKeyboardDownloadFinished`